src/lib.rs

```diff
@@ -0,0 +1,90 @@
+/*
+ *   Copyright (c) 2020 
+ *   All rights reserved.
+ */
+
+extern crate calamine;
+use calamine::{open_workbook, Reader, Xlsx};
+use std::fs::File;
+use std::io::BufReader;
+use std::io::Write;
+use std::string::String;
+
+pub fn create_file(input_file: &str) -> File {
+    let file_handle = File::create(input_file).expect("Error creating output file");
+    file_handle
+}
+
+pub fn open_file(output_file: &str) -> Xlsx<BufReader<File>> {
+    let workbook: Xlsx<_> = open_workbook(output_file).expect("Error while opening the input file");
+    workbook
+}
+
+pub fn operation(workbook: &mut Xlsx<BufReader<File>>, out_file: &mut File) {
+    let mut id = 1;
+    if let Some(Ok(reader)) = workbook.worksheet_range("Sheet1") {
+        for column in reader.rows().skip(1) {
+            let rate_index = column[1].to_string();
+            let combo = column[8].to_string();
+            let term_slab = column[6].to_string();
+            let seperated_str: Vec<&str> = combo.split("-").collect();
+            let from_amt = seperated_str[0];
+            let to_amt = seperated_str[1];
+            let mut seperated_space: Vec<&str> = term_slab.split(" ").collect();
+            let from_months: i32 = seperated_space[0]
+                .parse()
+                .expect("Error while parsing months to integer");
+            let from_year = from_months / 12;
+            let from_months = from_months % 12;
+            let from_days = seperated_space[2];
+            let to_year;
+            let mut to_months;
+            let to_days;
+            let mut month = String::new();
+            seperated_space.push("mon");
+            if seperated_space[4].to_uppercase() != "MONTHS" {
+                to_year = 10;
+                to_months = 0;
+                to_days = String::from("0");
+            } else {
+                for i in 0..(seperated_space[3].len()) {
+                    if seperated_space[3]
+                        .chars()
+                        .nth(i)
+                        .expect("Error while seperating month from alphanumeric")
+                        .is_numeric()
+                    {
+                        month.push(
+                            seperated_space[3]
+                                .chars()
+                                .nth(i)
+                                .expect("Error while pushing the seperated alphanumeric"),
+                        );
+                    }
+                }
+                to_months = (&month)
+                    .parse()
+                    .expect("Error while parsing months to integer");
+                to_days = seperated_space[5].to_string();
+                to_year = to_months / 12;
+                to_months = to_months % 12;
```

Comment:
all of these should be a separate module....do it then i will review the logic u have used